core/pool_vector.h

```diff
@@ -433,47 +526,70 @@ class PoolVector {
 
 	String join(String delimiter) {
 		String rs = "";
-		int s = size();
 		Read r = read();
+		int s = size();
 		for (int i = 0; i < s; i++) {
 			rs += r[i] + delimiter;
 		}
 		rs.erase(rs.length() - delimiter.length(), delimiter.length());
 		return rs;
 	}
 
-	bool is_locked() const { return alloc && alloc->lock.get() > 0; }
-
 	inline T operator[](int p_index) const;
 
 	Error resize(int p_size);
 
 	void invert();
 
 	void operator=(const PoolVector &p_pool_vector) { _reference(p_pool_vector); }
-	PoolVector() { alloc = nullptr; }
+	PoolVector() {
+		_alloc = memnew(Alloc);
+		_alloc->refcount.init(1);
+
+#ifdef DEBUG_ENABLED
+		MemoryPool::counter_mutex.lock();
+		MemoryPool::allocs_used++;
+		MemoryPool::total_memory += sizeof(Alloc);
+		if (MemoryPool::total_memory > MemoryPool::max_memory) {
+			MemoryPool::max_memory = MemoryPool::total_memory;
+		}
+		MemoryPool::counter_mutex.unlock();
+#endif
+
+#ifdef GODOT_POOL_VECTOR_REPORT_LEAKS
+		MemoryPool::report_alloc(_alloc, __LINE__);
+#endif
+	}
 	PoolVector(const PoolVector &p_pool_vector) {
-		alloc = nullptr;
+		_alloc = nullptr;
 		_reference(p_pool_vector);
 	}
-	~PoolVector() { _unreference(); }
+	~PoolVector() {
```

Comment:
This destructor should as well be virtual

Reply:
Hmmm..  Maybe I suppose maybe not, as it seems there are some size assertions with gdnative that fail with a vtable.

Reply:
I can't see an immediate reason why it needs to be virtual. Nothing is currently deriving from `PoolVector`.